File: app/src/main/java/mariavv/fitnesspal/presentation/journal/PagerAdapter.java
package mariavv.fitnesspal.presentation.journal;

import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import mariavv.fitnesspal.presentation.journal.daycard.DayCardFragment;

class PagerAdapter extends FragmentPagerAdapter {

    private long[] items;

    PagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int i) {
        return DayCardFragment.newInstance(items[i]);
    }

    @Override
    public int getCount() {
        if (items != null) {
            return items.length;
        }
        return 0;
    }

    void setItems(@NonNull long[] journalDaysCount) {
        items = journalDaysCount;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/mariavv/fitnesspal/presentation/journal/PagerAdapter.kt
package mariavv.fitnesspal.presentation.journal

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

import mariavv.fitnesspal.presentation.journal.daycard.DayCardFragment

internal class PagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private var items: ArrayList<Long>? = null

    override fun getItem(i: Int): Fragment {
        return DayCardFragment.newInstance(items!![i])
    }

    override fun getCount(): Int {
        return if (items != null) {
            items!!.size
        } else 0
    }

    fun setItems(journalDays: ArrayList<Long>) {
        items = journalDays
        notifyDataSetChanged()
    }

    fun add(position: Int, value: Long) {
        items?.add(position, value)
        notifyDataSetChanged()
    }
}
